File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        a, b, c = -1, -1, -1
        ans = 0
        
        for i in range(len(s)):
            if s[i] == 'a':
                a = i
            elif s[i] == 'b':
                b = i
            elif s[i] == 'c':
                c = i
            ans += min(a, b, c) + 1
        
        return ans

File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.go
func numberOfSubstrings(s string) int {
	lastPos := []int{-1, -1, -1}
	total := 0

	for pos, char := range s {
		lastPos[char-'a'] = pos
		total += 1 + min(lastPos[0], lastPos[1], lastPos[2])
	}

	return total
}